memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/MemeolistApplication.java

```diff
@@ -23,8 +23,10 @@ public void onCreate() {
         authService = new AuthService(authServiceConfig);
 
         if(authService.currentUser() != null) {
+            // FIXME This is serious bug (profile should change at any time and should not be cached)
             userProfile = new UserProfile(authService.currentUser());
         }
+	MobileCore.getInstance().requestHeaderInterceptor().add(authService.getAuthHeaderProvider());
```

Comment:
Just wondering if it should not be automatically handled by the SDK.

Reply:
I will like that but with currect core architecture where we are using single http instace per entire sdk is not going to work. Sync will also need dependency on auth. Do you mean that auth should do it automatically?

Problem here is that this token is also going to be send to other services like metrics that do not support authentication. Doing that automatically could be a problem for usages when Auth will be used with some external server etc.